src/ray/rpc/worker/direct_actor_client.h

```diff
@@ -40,13 +40,16 @@ class DirectActorClient : public std::enable_shared_from_this<DirectActorClient>
   /// \return if the rpc call succeeds
   ray::Status PushTask(std::unique_ptr<PushTaskRequest> request,
                        const ClientCallback<PushTaskReply> &callback) {
-    request->set_sequence_number(next_seq_no_++);
-    send_queue_.push_back(std::make_pair(std::move(request), callback));
+    request->set_sequence_number(request->task_spec().actor_task_spec().actor_counter());
+    {
+      std::lock_guard<std::mutex> lock(mutex_);
+      send_queue_.push_back(std::make_pair(std::move(request), callback));
+    }
```

Comment:
Does this make the entire client thread-safe (not sure if it we were assuming it was called from a single thread only before).

Reply:
Uhh I think so? Honestly not sure how this passed before. I ran into some bug here when testing the #5889 and added this to fix it.